### PR TITLE
Work to get ci/Travis working for windows

### PR DESCRIPTION
**Pull request by Mohjive**
This is WIP to make Travis work for Windows (git for Windows environment)

### CODE REVIEW

File: tests/_test_base.bash [hunk 1]
```diff
@@ -30,6 +30,12 @@ BEGIN { OFS=":"; FS=":"; }
 # This command is used with absolute homedir set and disabled warnings:
 GPGTEST="$SECRETS_GPG_COMMAND --homedir=$TEST_GPG_HOMEDIR --no-permission-warning --batch"
 
+# Platform stuff:
+if [[ "$GITSECRET_DIST" == "windows" ]]; then
+PS_CMD="ps -l -u";
+else
+PS_CMD="ps -wx -U";
```
Comment (joshrabinowitz) [inline tests/_test_base.bash]: can we indent these to be consistent with the rest of the file?

File: tests/_test_base.bash [hunk 2]
```diff
@@ -66,8 +72,8 @@ function test_user_password {
 
 function stop_gpg_agent {
   local username=$(id -u -n)
-  ps -wx -U "$username" | gawk \
-    '/gpg-agent --homedir/ { if ( $0 !~ "awk" ) { system("kill -9 "$1) } }' \
+  ${PS_CMD} "$username" | gawk \
+    '/gpg-agent --homedir/ { if ( $0 !~ "awk" ) { system("echo -9 "$1) } }' \
```
Comment (joshrabinowitz) [inline tests/_test_base.bash]: why this change from 'kill' to 'echo' ?

File: Makefile [hunk 3]
```diff
@@ -42,8 +42,8 @@ install-test:
 .PHONY: test
 test: install-test clean build
 	@chmod +x "./utils/tests.sh"; sync; \
-	export SECRET_PROJECT_ROOT="${PWD}"; \
-	export PATH="${PWD}/vendor/bats-core/bin:${PWD}:${PATH}"; \
+	export SECRET_PROJECT_ROOT="$(shell echo $${PWD})"; \
+	export PATH="$(shell echo $${PWD})/vendor/bats-core/bin:$(shell echo $${PWD}):$(shell echo $${PATH})"; \
```
Comment (joshrabinowitz) [inline Makefile]: can we add a comment to explain why we're going through these gyrations with PWD?

**Comment by joshrabinowitz** [comment]
Thank you @Mohjive ! Taking a look

File: Makefile (hunk 3 above)
Comment (Mohjive) [reply Makefile]: Added comments, see commit below

File: tests/_test_base.bash (hunk 2 above)
Comment (Mohjive) [reply tests/_test_base.bash]: Sorry, some lingering debug code. Fixed below

File: tests/_test_base.bash (hunk 1 above)
Comment (Mohjive) [reply tests/_test_base.bash]: Of course, see below.

File: Makefile (hunk 3 above)
Comment (joshrabinowitz) [reply Makefile]: Thank you!

Can we put this comment above the Makefile `.PHONY:test` line so it's not part of the test: command output?

Also maybe a comment like "the shell echo $${PWD}" construct is to get the Unix PATH env var on windows" would be clearer.

**Comment by joshrabinowitz** [comment]
I was reading about msys2 and came across references to 'MSYS2_PATH_TYPE=inherit' which I read as making the msys2 PATH match the windows one.

See "Get Windows Environment to msys2" at https://juniway.github.io/2016/08/26/linux-msys2/ - I couldn't find better docs about this.

I seem to recall some other method of synchronizing env vars (PATH and others) between windows and msys2 but I can't seem to dig up references now.

**Comment by Mohjive** [comment]
I'll look into that.

I did find this when looking into not mangling paths the other day: https://stackoverflow.com/a/34386471, but none of those were working for this issue.

**Comment by Mohjive** [comment]
Seems that **MSYS2_PATH_TYPE=inherit** is used to control wether windows PATH is exposed into MSYS environment, but not how paths are handled.

**Comment by Mohjive** [comment]
Closing this with reference to #375. All work is being done there.
